src/ledger/InternalLedgerEntry.cpp

```diff
@@ -131,10 +132,39 @@ InternalLedgerKey::~InternalLedgerKey()
     destruct();
 }
 
+size_t
+InternalLedgerKey::hash() const
+{
+    if (mHash != 0)
+    {
+        return mHash;
+    }
+    size_t res;
+    switch (type())
```

Comment:
I know this is the hash function that we were already using, but it's not very good. It collides sponsorships and sponsorship counters.